packages/eslint-plugin/src/rules/indent.ts

```diff
@@ -340,7 +340,26 @@ export default util.createRule<Options, MessageIds>({
           loc: node.loc,
         });
       },
+      'FunctionDeclaration, FunctionExpression'(
+        node: TSESTree.FunctionDeclaration,
+      ) {
+        // https://github.com/typescript-eslint/typescript-eslint/issues/1649
+        if (node.params) {
+          for (const p of node.params) {
+            if (p.type !== AST_NODE_TYPES.Identifier) {
+              continue;
+            }
+
+            if (p.decorators && p.decorators.length) {
+              p.loc.start = p.decorators[0].loc.start;
+              p.range[0] = p.decorators[0].range[0];
+              p.decorators = [];
+            }
```

Comment:
mutating the node is not something you should do.
The AST is shared amongst all of the rules, so mutating the AST node itself can affect the entire lint run.